src/trace/dma-trace.c

```diff
@@ -311,6 +324,14 @@ static int dma_trace_start(struct dma_trace_data *d)
 			d->dc.chan = NULL;
 			err = dma_copy_set_stream_tag(&d->dc, d->stream_tag);
 		}
+
+		/* Reset host buffer information as host is re-configuring dtrace */
+		spin_lock_irq(&d->lock, flags);
+
+		d->old_host_offset = 0;
+		d->posn.host_offset = 0;
```

Comment:
above `dma_stop()` or `dma_copy_set_stream_tag()` could've failed. Don't you need to abort in some of those cases? I'd reformat this function with the usual `goto`s to make error handling more obvious

Reply:
@lyakh, we do abort at line 339.

This is a bit tricky. I would say that we can (and need) to reset the the host_offset in any case, so this can be moved beyond line 341.

Reply:
Moved after the error check/abort.